test.disabled/ejabberd_tests/tests/sm_SUITE.erl

```diff
@@ -791,11 +791,26 @@ assert_no_offline_msgs(Spec) ->
 assert_no_offline_msgs() ->
     0 = mongoose_helper:total_offline_messages().
 
+wait_for_c2s_state_change(C2SPid, StateName, NewStateName) ->
+    wait_for_c2s_state_change(C2SPid, StateName, NewStateName, 5000).
+
+wait_for_c2s_state_change(C2SPid, StateName, NewStateName, 0) ->
```

Comment:
This is error prone. We may end up in an infinite loop if someone wants to wait `5001` ms for instance.